src/parquet/util/input.cc

```diff
@@ -134,7 +134,10 @@ void MemoryMapSource::Close() {
 }
 
 void MemoryMapSource::CloseFile() {
-  if (data_ != nullptr) { munmap(data_, size_); }
+  if (data_ != nullptr) {
+    munmap(data_, size_);
+    data_ = nullptr;
+  }
```

Comment:
It seems reasonable to add a test for idempotency. Calling `MemoryMapSource::Close` twice should trigger the issue?